wasm/src/lib.rs

```diff
@@ -146,7 +146,7 @@ pub fn query_triggering_event() -> Event {
 /// # Traps
 ///
 /// Host side will generate a trap if this function was not called from a validator.
-pub fn query_operation_to_validate() -> NeedsValidationBox {
+pub fn query_operation_to_validate() -> Option<NeedsValidationBox> {
```

Comment:
Because previously we used to just return `Trap` if `State` does not contains such operation.
And I decided to make this handling on user-side. Like recoverable error

Reply:
I'm not sure I agree with this. This indicates that smart contract is incorrectly implemented or that we have a bug in Iroha. This is not recoverable IMO

The only way I can think of how someone could use this is to have the same smart contract for both trigger and validation functionality. I don't find this very useful

Reply:
Ok, let's make it a `Trap`

Reply:
Or panic

Reply:
Fixed